haskell-api.md

```diff
@@ -117,6 +117,98 @@ possible. Issues include:
 If you are creating a new `PUT` consider if it could be expressed in a more
 granular `PATCH` semantic.
 
+### `PATCH` semantics
+
+Modifications to an existing resource should be made via a `PATCH` request.
+
+### Specific guidelines
+
+A well-formed `PATCH` endpoint
+
+- Must update an entity's `updatedAt` field, if present.
+- Must return an empty response or the modified resource. If the latter, the
```

Comment:
What is the argument for empty response being allowed?

Reply:
Personally I'm in favor removing the `empty` affordance and just saying we must return the modified resource. But if I had to contrive a few reasons for allowing empty they'd be
 - easier/simpler implementation
 - forces consumer to re-fetch or maintain state, so it avoids the "whoops my patch returned an inconsistent object" (e.g. made updates in-memory and forgot to persist)
 - potentially faster implementation (no refetch is necessary)

Reply:
Gotcha. Given that we only have contrived reasons, it feels more appropriate to downgrade this to SHOULD without the `empty` affordance. That way, cases that do need it won't be violating a MUST, and we can discover them and update the guide if necessary based on concrete cases. The way it's written now, an empty response is a first-class option that some may choose without any nudge the other way -- which, I think, is not what we'd want ideally-speaking. Right?

That said, I Approved along with this question so I'm also fine keeping it as-is.

Reply:
Should w/o `empty` is good with me!